src/middleware/login.middleware.js

```diff
@@ -0,0 +1,27 @@
+const SToken = require('../services/token.service');
+const M = require('../models');
+
+/**
+ * Add to the req.infos, the prototype of the student or teacher who make the request.
+ */
+const login = async function (req, res, next) {
+  const token = req.headers.authorization;
+  if (token) {
+    const infos = SToken.decryptToken(req.headers.authorization);
+    let user = null;
+    if (infos.role === 'student') {
+      user = await M.Student.getByName(infos.firstname, infos.lastname); 
+    } else if (infos.role === 'teacher') {
+      user = await M.Teacher.getByName(infos.firstname, infos.lastname);
```

Comment:
What if firsname, lastname are not unique ?

Reply:
It’s unique, thanks to UM connection.

Reply:
I did not know that. Great !

Reply:
Does this mean if we have 2 people called "John Doe", we would have this:

* John Doe
* John Doe2